### PR TITLE
Add NuGet config and update autorest.testserver version

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes - https://github.com/Azure/autorest.csharp/issues/820

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -4,7 +4,6 @@
     <RepoEngPath>$(MSBuildThisFileDirectory)eng</RepoEngPath>
     <RepoSrcPath>$(RepoRoot)src</RepoSrcPath>
     <IsTestProject Condition="$(MSBuildProjectName.EndsWith('.Tests'))">true</IsTestProject>
-    <RestoreAdditionalProjectSources>https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/index.json</RestoreAdditionalProjectSources>
```
Comment (pakrym) [inline Directory.Build.props:7]: This source should be added to nuget.config

File: NuGet.Config [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="Local" value="tools/LocalNugetFeed" />
```
Comment (pakrym) [inline NuGet.Config:5]: I think we can remove this one

File: NuGet.Config [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    <add key="azure-sdk-tools" value="https://azuresdkartifacts.blob.core.windows.net/azure-sdk-tools/index.json" />
+    <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
```
Comment (pakrym) [inline NuGet.Config:7]: We don't use these ones.

File: Directory.Build.props (hunk 1 above)
Comment (ShivangiReja) [reply Directory.Build.props:7]: Isn't it the same as `<add key="azure-sdk-tools" value="https://azuresdkartifacts.blob.core.windows.net/azure-sdk-tools/index.json" />`?

Comment (ShivangiReja) [reply Directory.Build.props:7]: I accidentally substituted this with `https://azuresdkartifacts.blob.core.windows.net/azure-sdk-tools/index.json`.

Comment (ShivangiReja) [reply Directory.Build.props:7]: Yeah, I accidentally substituted this with `https://azuresdkartifacts.blob.core.windows.net/azure-sdk-tools/index.json`.

Added this to NuGet.Config.

Comment (ShivangiReja) [reply Directory.Build.props:7]: Yeah, my bad I accidentally substituted this with `https://azuresdkartifacts.blob.core.windows.net/azure-sdk-tools/index.json`.

Added this to NuGet.Config.

File: NuGet.Config [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="azure-sdk-for-net" value="https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/index.json" />
```
Comment (weshaggard) [inline NuGet.Config:5]: I suggest using the Azure Devops feed like .NET, see https://github.com/Azure/azure-sdk-for-net/blob/master/NuGet.Config#L8.

Comment (weshaggard) [reply NuGet.Config:5]: Sorry I just realized that the .NET nuget.config doesn't have the devops feed yet either. The feed I suggest using is 
```
<add key="azure-sdk-for-net" value="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json" />
```

Comment (weshaggard) [reply NuGet.Config:5]: .NET repo update is still in PR https://github.com/Azure/azure-sdk-for-net/pull/11916.

Comment (ShivangiReja) [reply NuGet.Config:5]: Fixed here - https://github.com/Azure/autorest.csharp/pull/826

**Pull request merged**